### PR TITLE
Reduce uses of `eval` for regex literal syntax

### LINKED ISSUE #1882: regex named groups tests using eval
The tests in the following files are checking for syntax errors in regex literals using `eval`:
- [non-unicode-malformed.js](https://github.com/tc39/test262/blob/3936950114fcdf49ece33bbbc52b71ac4ebbb32c/test/built-ins/RegExp/named-groups/non-unicode-malformed.js)
- [non-unicode-property-names.js](https://github.com/tc39/test262/blob/3936950114fcdf49ece33bbbc52b71ac4ebbb32c/test/built-ins/RegExp/named-groups/non-unicode-property-names.js)
- [unicode-malformed.js](https://github.com/tc39/test262/blob/3936950114fcdf49ece33bbbc52b71ac4ebbb32c/test/built-ins/RegExp/named-groups/unicode-malformed.js)
- [unicode-property-names.js](https://github.com/tc39/test262/blob/3936950114fcdf49ece33bbbc52b71ac4ebbb32c/test/built-ins/RegExp/named-groups/unicode-property-names.js)

This makes them much less useful for parsers (ref #1356). Would y'all accept a PR splitting each negative test from these into its own file with the standard parse-error frontmatter?

(cc @jugglinmike)

### PR DESCRIPTION
**Pull request by bakkot**
Fixes #1882.

Some notes:
- I went through each of the four files referenced in that issue and manually split tests out. Each problematic file has its own commit.
- For positive tests, I left the original files where they are, but it might make more sense for them to live in `langauge/literals/regexp/named-groups` rather than `built-ins/RegExp/named-groups/`.
- Descriptions and, occasionally, info blocks are best-effort based on the original tests. As such, they're sometimes a bit lacking. Still it is a strict improvement over the existing state, I think.
- I have not verified correctness of these tests in any way. I just copied the original tests.

### CODE REVIEW

File: test/built-ins/RegExp/named-groups/non-unicode-property-names.js [hunk 1]
```diff
@@ -5,40 +5,19 @@
 description: Exotic named group names in non-Unicode RegExps
 esid: prod-GroupSpecifier
 features: [regexp-named-groups]
-includes: [compareArray.js]
 ---*/
 
 assert.sameValue("a", /(?<π>a)/.exec("bab").groups.π);
-assert.throws(SyntaxError, () => eval('/(?<\\u{03C0}>a)/'), "\\u{} escapes allowed only in Unicode mode");
```
Comment (rwaldron) [inline test/built-ins/RegExp/named-groups/non-unicode-property-names.js]: Should there now be a standalone test for this?

File: test/built-ins/RegExp/named-groups/non-unicode-property-names.js [hunk 2]
```diff
@@ -5,40 +5,19 @@
...
 assert.sameValue("a", /(?<_>a)/.exec("bab").groups._);
-assert.throws(SyntaxError, () => eval('/(?<$𐒤>a)/'), "Individual surrogates not in ID_Continue");
 assert.sameValue("a", /(?<_\u200C>a)/.exec("bab").groups._\u200C);
 assert.sameValue("a", /(?<_\u200D>a)/.exec("bab").groups._\u200D);
 assert.sameValue("a", /(?<ಠ_ಠ>a)/.exec("bab").groups.ಠ_ಠ);
-assert.throws(SyntaxError, () => eval('/(?<❤>a)/'));
-assert.throws(SyntaxError, () => eval('/(?<𐒤>a)/'), "Individual surrogate not in ID_Start.");
 
 // Unicode escapes in capture names.
-assert.throws(SyntaxError, () => eval("/(?<a\\uD801\uDCA4>.)/"));
```
Comment (rwaldron) [inline test/built-ins/RegExp/named-groups/non-unicode-property-names.js]: Should there now be a standalone test for this?

**Comment by bakkot** [comment]
@rwaldron Hm, I didn't mean to leave things out, I don't think, except for one or two duplicates.

I'll go through and link tests in the original files to their corresponding files for ease of comparison. Give me a bit.

File: test/built-ins/RegExp/named-groups/unicode-property-names.js [hunk 3]
```diff
@@ -17,30 +17,12 @@ assert.sameValue("a", /(?<$𐒤>a)/u.exec("bab").groups.$𐒤);
 assert.sameValue("a", /(?<_\u200C>a)/u.exec("bab").groups._\u200C);
 assert.sameValue("a", /(?<_\u200D>a)/u.exec("bab").groups._\u200D);
 assert.sameValue("a", /(?<ಠ_ಠ>a)/u.exec("bab").groups.ಠ_ಠ);
-assert.throws(SyntaxError, () => eval('/(?<❤>a)/u'));
-assert.throws(SyntaxError, () => eval('/(?<𐒤>a)/u'), "ID_Continue but not ID_Start.");
 
 // Unicode escapes in capture names.
 assert(/(?<a\uD801\uDCA4>.)/u.test("a"), "\\u Lead \\u Trail");
-assert.throws(SyntaxError, () => eval("/(?<a\\uD801>.)/u"), "\\u Lea");
```
Comment (rwaldron) [inline test/built-ins/RegExp/named-groups/unicode-property-names.js]: Missing standalone test

File: test/built-ins/RegExp/named-groups/non-unicode-property-names.js [hunk 4]
```diff
@@ -5,40 +5,19 @@
...
-assert.throws(SyntaxError, () => eval("/(?<a\\uD801>.)/"));
```
Comment (rwaldron) [inline test/built-ins/RegExp/named-groups/non-unicode-property-names.js]: Missing standalone test

File: test/built-ins/RegExp/named-groups/unicode-property-names.js [hunk 5]
```diff
@@ -17,30 +17,12 @@ assert.sameValue("a", /(?<$𐒤>a)/u.exec("bab").groups.$𐒤);
...
-assert.throws(SyntaxError, () => eval("/(?<a\\uDCA4>.)/u"), "\\u Trai");
```
Comment (rwaldron) [inline test/built-ins/RegExp/named-groups/unicode-property-names.js]: Missing standalone test

File: test/built-ins/RegExp/named-groups/non-unicode-property-names.js [hunk 6]
```diff
@@ -5,40 +5,19 @@
...
-assert.throws(SyntaxError, () => eval("/(?<a\\uDCA4>.)/"));
```
Comment (rwaldron) [inline test/built-ins/RegExp/named-groups/non-unicode-property-names.js]: Missing standalone test

File: test/built-ins/RegExp/named-groups/non-unicode-property-names.js [hunk 7]
```diff
@@ -5,40 +5,19 @@
...
 assert(/(?<\u0041>.)/.test("a"));
-assert.throws(SyntaxError, () => eval("/(?<a\\u{104A4}>.)/"));
```
Comment (rwaldron) [inline test/built-ins/RegExp/named-groups/non-unicode-property-names.js]: Missing standalone test

File: test/built-ins/RegExp/named-groups/non-unicode-property-names.js [hunk 8]
```diff
@@ -5,40 +5,19 @@
...
-assert.throws(SyntaxError, () => eval("/(?<a\\u{10FFFF}>.)/"));
```
Comment (rwaldron) [inline test/built-ins/RegExp/named-groups/non-unicode-property-names.js]: Missing standalone test

File: test/built-ins/RegExp/named-groups/unicode-property-names.js [hunk 9]
```diff
@@ -17,30 +17,12 @@ assert.sameValue("a", /(?<$𐒤>a)/u.exec("bab").groups.$𐒤);
...
-assert.throws(SyntaxError, () => eval("/(?<a\\u{110000}>.)/u"), "\\u{ Out-of-bounds ");
-assert.throws(SyntaxError, () => eval("/(?<a\uD801>.)/u"), "Lea");
-assert.throws(SyntaxError, () => eval("/(?<a\uDCA4>.)/u"), "Trai");
-assert(RegExp("(?<\u{0041}>.)", "u").test("a"), "Non-surrogate");
-assert(RegExp("(?<a\u{104A4}>.)", "u").test("a"), "Surrogate,ID_Continue");
-
-// Bracketed escapes are not allowed;
-// 4-char escapes must be the proper ID_Start/ID_Continue
-assert.throws(SyntaxError, () => eval("/(?<a\\uD801>.)/u"), "Lead");
-assert.throws(SyntaxError, () => eval("/(?<a\\uDCA4>.)/u"), "Trail");
```
Comment (rwaldron) [inline test/built-ins/RegExp/named-groups/unicode-property-names.js]: Missing standalone test

File: test/built-ins/RegExp/named-groups/unicode-property-names.js [hunk 10]
```diff
@@ -17,30 +17,12 @@ assert.sameValue("a", /(?<$𐒤>a)/u.exec("bab").groups.$𐒤);
...
 assert((/(?<\u{0041}>.)/u).test("a"), "Non-surrogate");
 assert(/(?<a\u{104A4}>.)/u.test("a"), "Surrogate, ID_Continue");
 assert(RegExp("(?<\\u0041>.)", "u").test("a"), "Non-surrogate");
-
-// Backslash is not allowed as ID_Start and ID_Continue
-assert.throws(SyntaxError, () => eval("/(?<\\>.)/u"), "'\' misclassified as ID_Start");
```
Comment (rwaldron) [inline test/built-ins/RegExp/named-groups/unicode-property-names.js]: Missing standalone test

File: test/built-ins/RegExp/named-groups/non-unicode-property-names.js [hunk 11]
```diff
@@ -5,40 +5,19 @@
...
-// Bracketed escapes are not allowed;
 // 4-char escapes must be the proper ID_Start/ID_Continue
-assert.throws(SyntaxError, () => eval("/(?<a\\uD801>.)/"), "Lead");
-assert.throws(SyntaxError, () => eval("/(?<a\\uDCA4>.)/"), "Trail");
-assert.throws(SyntaxError, () => eval("/(?<\\u{0041}>.)/"), "Non-surrogate");
-assert.throws(SyntaxError, () => eval("/(?<a\\u{104A4}>.)/"), "Surrogate, ID_Continue");
 assert(RegExp("(?<\\u0041>.)").test("a"), "Non-surrogate");
-
-// Backslash is not allowed as ID_Start and ID_Continue
-assert.throws(SyntaxError, () => eval("/(?<\\>.)/"), "'\' misclassified as ID_Start");
```
Comment (rwaldron) [inline test/built-ins/RegExp/named-groups/non-unicode-property-names.js]: Missing standalone test

File: test/built-ins/RegExp/named-groups/unicode-property-names.js [hunk 12]
```diff
@@ -17,30 +17,12 @@ assert.sameValue("a", /(?<$𐒤>a)/u.exec("bab").groups.$𐒤);
...
 assert(/(?<a\u{104A4}>.)/u.test("a"), "\\u{ Surrogate, ID_Continue }");
-assert.throws(SyntaxError, () => eval("/(?<a\\u{110000}>.)/u"), "\\u{ Out-of-bounds ");
-assert.throws(SyntaxError, () => eval("/(?<a\uD801>.)/u"), "Lea");
-assert.throws(SyntaxError, () => eval("/(?<a\uDCA4>.)/u"), "Trai");
-assert(RegExp("(?<\u{0041}>.)", "u").test("a"), "Non-surrogate");
-assert(RegExp("(?<a\u{104A4}>.)", "u").test("a"), "Surrogate,ID_Continue");
-
-// Bracketed escapes are not allowed;
-// 4-char escapes must be the proper ID_Start/ID_Continue
-assert.throws(SyntaxError, () => eval("/(?<a\\uD801>.)/u"), "Lead");
```
Comment (rwaldron) [inline test/built-ins/RegExp/named-groups/unicode-property-names.js]: Missing standalone test

File: test/built-ins/RegExp/named-groups/unicode-property-names.js [hunk 13]
```diff
@@ -17,30 +17,12 @@ assert.sameValue("a", /(?<$𐒤>a)/u.exec("bab").groups.$𐒤);
...
-assert.throws(SyntaxError, () => eval('/(?<𐒤>a)/u'), "ID_Continue but not ID_Start.");
 
 // Unicode escapes in capture names.
 assert(/(?<a\uD801\uDCA4>.)/u.test("a"), "\\u Lead \\u Trail");
-assert.throws(SyntaxError, () => eval("/(?<a\\uD801>.)/u"), "\\u Lea");
-assert.throws(SyntaxError, () => eval("/(?<a\\uDCA4>.)/u"), "\\u Trai");
 assert(/(?<\u0041>.)/u.test("a"), "\\u NonSurrogate");
 assert(/(?<\u{0041}>.)/u.test("a"), "\\u{ Non-surrogate }");
 assert(/(?<a\u{104A4}>.)/u.test("a"), "\\u{ Surrogate, ID_Continue }");
-assert.throws(SyntaxError, () => eval("/(?<a\\u{110000}>.)/u"), "\\u{ Out-of-bounds ");
```
Comment (rwaldron) [inline test/built-ins/RegExp/named-groups/unicode-property-names.js]: Missing standalone test

File: test/built-ins/RegExp/named-groups/unicode-property-names.js [hunk 14]
```diff
@@ -17,30 +17,12 @@ assert.sameValue("a", /(?<$𐒤>a)/u.exec("bab").groups.$𐒤);
...
-// 4-char escapes must be the proper ID_Start/ID_Continue
-assert.throws(SyntaxError, () => eval("/(?<a\\uD801>.)/u"), "Lead");
-assert.throws(SyntaxError, () => eval("/(?<a\\uDCA4>.)/u"), "Trail");
 assert((/(?<\u{0041}>.)/u).test("a"), "Non-surrogate");
 assert(/(?<a\u{104A4}>.)/u.test("a"), "Surrogate, ID_Continue");
 assert(RegExp("(?<\\u0041>.)", "u").test("a"), "Non-surrogate");
-
-// Backslash is not allowed as ID_Start and ID_Continue
-assert.throws(SyntaxError, () => eval("/(?<\\>.)/u"), "'\' misclassified as ID_Start");
-assert.throws(SyntaxError, () => eval("/(?<a\\>.)/u"), "'\' misclassified as ID_Continue");
```
Comment (rwaldron) [inline test/built-ins/RegExp/named-groups/unicode-property-names.js]: Missing standalone test

**Comment by rwaldron** [comment]
@bakkot I just went through every removed test and searched for its new standalone file, I marked all of the removed tests that are missing a standalone test file.

**Comment by bakkot** [comment]
@rwaldron The majority of those do have a corresponding standalone test file. I expect you were missing them because the original tests have escaped backslashes whereas the standalone test do not. However, there were a small number which were in fact missing, which I've added as a new commit.

Here's a list of every regex this PR removes from an existing file, in order of appearance, with a link to the new file where that test now resides:

from [non-unicode-malformed.js](https://github.com/tc39/test262/blob/69c1efd325deedf54db92a23008399e2b00fa51e/test/built-ins/RegExp/named-groups/non-unicode-malformed.js)

- [`/(?<>a)/`](https://github.com/tc39/test262/blob/0160cab5185232d0f7ada295b90fa687212a22bf/test/language/literals/regexp/named-groups/invalid-empty-groupspecifier.js)
- [`/(?<aa)/`](https://github.com/tc39/test262/blob/0160cab5185232d0f7ada295b90fa687212a22bf/test/language/literals/regexp/named-groups/invalid-unterminated-groupspecifier.js)
- [`/(?<42a>a)/`](https://github.com/tc39/test262/blob/0160cab5185232d0f7ada295b90fa687212a22bf/test/language/literals/regexp/named-groups/invalid-numeric-groupspecifier.js)
- [`/(?<:a>a)/`](https://github.com/tc39/test262/blob/0160cab5185232d0f7ada295b90fa687212a22bf/test/language/literals/regexp/named-groups/invalid-punctuator-starting-groupspecifier.js)
- [`/(?<a:>a)/`](https://github.com/tc39/test262/blob/0160cab5185232d0f7ada295b90fa687212a22bf/test/language/literals/regexp/named-groups/invalid-punctuator-within-groupspecifier.js)
- [`/(?<a>a)(?<a>a)/`](https://github.com/tc39/test262/blob/0160cab5185232d0f7ada295b90fa687212a22bf/test/language/literals/regexp/named-groups/invalid-duplicate-groupspecifier.js)
- [`/(?<a>a)(?<b>b)(?<a>a)/`](https://github.com/tc39/test262/blob/0160cab5185232d0f7ada295b90fa687212a22bf/test/language/literals/regexp/named-groups/invalid-duplicate-groupspecifier-2.js)
- [`/(?<a>.)\\k/`](https://github.com/tc39/test262/blob/0160cab5185232d0f7ada295b90fa687212a22bf/test/language/literals/regexp/named-groups/invalid-incomplete-groupname.js)
- [`/(?<a>.)\\k<a/`](https://github.com/tc39/test262/blob/0160cab5185232d0f7ada295b90fa687212a22bf/test/language/literals/regexp/named-groups/invalid-incomplete-groupname-2.js)
- [`/(?<a>.)\\k<>/`](https://github.com/tc39/test262/blob/0160cab5185232d0f7ada295b90fa687212a22bf/test/language/literals/regexp/named-groups/invalid-incomplete-groupname-3.js)
- [`/(?<a>.)\\k<b>/`](https://github.com/tc39/test262/blob/0160cab5185232d0f7ada295b90fa687212a22bf/test/language/literals/regexp/named-groups/invalid-dangling-groupname.js)
- [`/(?<a>a)\\k<ab>/`](https://github.com/tc39/test262/blob/0160cab5185232d0f7ada295b90fa687212a22bf/test/language/literals/regexp/named-groups/invalid-dangling-groupname-2.js)
- [`/(?<ab>a)\\k<a>/`](https://github.com/tc39/test262/blob/0160cab5185232d0f7ada295b90fa687212a22bf/test/language/literals/regexp/named-groups/invalid-dangling-groupname-3.js)
- [`/\\k<a>(?<ab>a)/`](https://github.com/tc39/test262/blob/0160cab5185232d0f7ada295b90fa687212a22bf/test/language/literals/regexp/named-groups/invalid-dangling-groupname-4.js)
- [`/\\k<a(?<a>a)/`](https://github.com/tc39/test262/blob/0160cab5185232d0f7ada295b90fa687212a22bf/test/language/literals/regexp/named-groups/invalid-incomplete-groupname-4.js)
- [`/\k<a>(?<a>x)/`](https://github.com/tc39/test262/blob/0160cab5185232d0f7ada295b90fa687212a22bf/test/language/literals/regexp/named-groups/forward-reference.js)
- [`/\\k<a>(?<b>x)/`](https://github.com/tc39/test262/blob/0160cab5185232d0f7ada295b90fa687212a22bf/test/language/literals/regexp/named-groups/invalid-dangling-groupname-5.js)
- [`/\\k<a(?<a>.)/`](https://github.com/tc39/test262/blob/0160cab5185232d0f7ada295b90fa687212a22bf/test/language/literals/regexp/named-groups/invalid-incomplete-groupname-5.js)
- [`/\\k(?<a>.)/`](https://github.com/tc39/test262/blob/0160cab5185232d0f7ada295b90fa687212a22bf/test/language/literals/regexp/named-groups/invalid-incomplete-groupname-6.js)


from [non-unicode-property-names.js](https://github.com/tc39/test262/blob/69c1efd325deedf54db92a23008399e2b00fa51e/test/built-ins/RegExp/named-groups/non-unicode-property-names.js)

- [`/(?<\\u{03C0}>a)/`](https://github.com/tc39/test262/blob/0160cab5185232d0f7ada295b90fa687212a22bf/test/language/literals/regexp/named-groups/invalid-u-escape-in-groupspecifier-2.js)
- [`/(?<$𐒤>a)/`](https://github.com/tc39/test262/blob/0160cab5185232d0f7ada295b90fa687212a22bf/test/language/literals/regexp/named-groups/invalid-non-id-continue-groupspecifier.js)
- [`/(?<❤>a)/`](https://github.com/tc39/test262/blob/0160cab5185232d0f7ada295b90fa687212a22bf/test/language/literals/regexp/named-groups/invalid-non-id-start-groupspecifier.js)
- [`/(?<𐒤>a)/`](https://github.com/tc39/test262/blob/0160cab5185232d0f7ada295b90fa687212a22bf/test/language/literals/regexp/named-groups/invalid-non-id-start-groupspecifier-2.js)
- [`/(?<a\\uD801\uDCA4>.)/`](https://github.com/tc39/test262/blob/0160cab5185232d0f7ada295b90fa687212a22bf/test/language/literals/regexp/named-groups/invalid-non-id-start-groupspecifier-3.js)
- [`/(?<a\\uD801>.)/`](https://github.com/tc39/test262/blob/0160cab5185232d0f7ada295b90fa687212a22bf/test/language/literals/regexp/named-groups/invalid-non-id-start-groupspecifier-4.js)
- [`/(?<a\\uDCA4>.)/`](https://github.com/tc39/test262/blob/0160cab5185232d0f7ada295b90fa687212a22bf/test/language/literals/regexp/named-groups/invalid-non-id-start-groupspecifier-5.js)
- [`/(?<a\\u{104A4}>.)/`](https://github.com/tc39/test262/blob/0160cab5185232d0f7ada295b90fa687212a22bf/test/language/literals/regexp/named-groups/invalid-non-id-start-groupspecifier-6.js)
- [`/(?<a\\u{10FFFF}>.)/`](https://github.com/tc39/test262/blob/0160cab5185232d0f7ada295b90fa687212a22bf/test/language/literals/regexp/named-groups/invalid-non-id-start-groupspecifier-7.js)
- [`/(?<a\uD801>.)/`](https://github.com/tc39/test262/blob/0160cab5185232d0f7ada295b90fa687212a22bf/test/language/literals/regexp/named-groups/invalid-lone-surrogate-groupname.js)
- [`/(?<a\uDCA4>.)/`](https://github.com/tc39/test262/blob/0160cab5185232d0f7ada295b90fa687212a22bf/test/language/literals/regexp/named-groups/invalid-lone-surrogate-groupname.js)
- `/(?<a\\uD801>.)/` DUP IN ORIGINAL
- `/(?<a\\uDCA4>.)/` DUP IN ORIGINAL
- [`/(?<\\u{0041}>.)/`](https://github.com/tc39/test262/blob/0160cab5185232d0f7ada295b90fa687212a22bf/test/language/literals/regexp/named-groups/invalid-u-escape-in-groupspecifier.js)
- /(?<a\\u{104A4}>.)/ DUP IN ORIGINAL
- [`/(?<\\>.)/`](https://github.com/tc39/test262/blob/0160cab5185232d0f7ada295b90fa687212a22bf/test/language/literals/regexp/named-groups/invalid-non-id-start-groupspecifier-8.js)
- [`/(?<a\\>.)/`](https://github.com/tc39/test262/blob/0160cab5185232d0f7ada295b90fa687212a22bf/test/language/literals/regexp/named-groups/invalid-non-id-continue-groupspecifier-4.js)


from [unicode-malformed.js](https://github.com/tc39/test262/blob/69c1efd325deedf54db92a23008399e2b00fa51e/test/built-ins/RegExp/named-groups/unicode-malformed.js)

- [`/(?<>a)/u`](https://github.com/tc39/test262/blob/0160cab5185232d0f7ada295b90fa687212a22bf/test/language/literals/regexp/named-groups/invalid-empty-groupspecifier-u.js)
- [`/(?<aa)/u`](https://github.com/tc39/test262/blob/0160cab5185232d0f7ada295b90fa687212a22bf/test/language/literals/regexp/named-groups/invalid-unterminated-groupspecifier-u.js)
- [`/(?<42a>a)/u`](https://github.com/tc39/test262/blob/0160cab5185232d0f7ada295b90fa687212a22bf/test/language/literals/regexp/named-groups/invalid-numeric-groupspecifier-u.js)
- [`/(?<:a>a)/u`](https://github.com/tc39/test262/blob/0160cab5185232d0f7ada295b90fa687212a22bf/test/language/literals/regexp/named-groups/invalid-punctuator-starting-groupspecifier-u.js)
- [`/(?<a:>a)/u`](https://github.com/tc39/test262/blob/0160cab5185232d0f7ada295b90fa687212a22bf/test/language/literals/regexp/named-groups/invalid-punctuator-within-groupspecifier-u.js)
- [`/(?<a>a)(?<a>a)/u`](https://github.com/tc39/test262/blob/0160cab5185232d0f7ada295b90fa687212a22bf/test/language/literals/regexp/named-groups/invalid-duplicate-groupspecifier-u.js)
- [`/(?<a>a)(?<b>b)(?<a>a)/u`](https://github.com/tc39/test262/blob/0160cab5185232d0f7ada295b90fa687212a22bf/test/language/literals/regexp/named-groups/invalid-duplicate-groupspecifier-2-u.js)
- [`/\\k<a>/u`](https://github.com/tc39/test262/blob/0160cab5185232d0f7ada295b90fa687212a22bf/test/language/literals/regexp/named-groups/invalid-dangling-groupname-without-group-u.js)
- [`/\\k<a/u`](https://github.com/tc39/test262/blob/0160cab5185232d0f7ada295b90fa687212a22bf/test/language/literals/regexp/named-groups/invalid-incomplete-groupname-without-group-u.js)
- [`/\\k<>/u`](https://github.com/tc39/test262/blob/0160cab5185232d0f7ada295b90fa687212a22bf/test/language/literals/regexp/named-groups/invalid-incomplete-groupname-without-group-2-u.js)
- [`/\\k/u`](https://github.com/tc39/test262/blob/0160cab5185232d0f7ada295b90fa687212a22bf/test/language/literals/regexp/named-groups/invalid-incomplete-groupname-without-group-3-u.js)
- [`/(?<a>.)\\k/u`](https://github.com/tc39/test262/blob/0160cab5185232d0f7ada295b90fa687212a22bf/test/language/literals/regexp/named-groups/invalid-incomplete-groupname-u.js)
- [`/(?<a>.)\\k<a/u`](https://github.com/tc39/test262/blob/0160cab5185232d0f7ada295b90fa687212a22bf/test/language/literals/regexp/named-groups/invalid-incomplete-groupname-2-u.js)
- [`/(?<a>.)\\k<>/u`](https://github.com/tc39/test262/blob/0160cab5185232d0f7ada295b90fa687212a22bf/test/language/literals/regexp/named-groups/invalid-incomplete-groupname-3-u.js)
- [`/(?<a>.)\\k<b>/u`](https://github.com/tc39/test262/blob/0160cab5185232d0f7ada295b90fa687212a22bf/test/language/literals/regexp/named-groups/invalid-dangling-groupname-u.js)
- [`/(?<a>a)\\k<ab>/u`](https://github.com/tc39/test262/blob/0160cab5185232d0f7ada295b90fa687212a22bf/test/language/literals/regexp/named-groups/invalid-dangling-groupname-2-u.js)
- [`/(?<ab>a)\\k<a>/u`](https://github.com/tc39/test262/blob/0160cab5185232d0f7ada295b90fa687212a22bf/test/language/literals/regexp/named-groups/invalid-dangling-groupname-3-u.js)
- [`/\\k<a>(?<ab>a)/u`](https://github.com/tc39/test262/blob/0160cab5185232d0f7ada295b90fa687212a22bf/test/language/literals/regexp/named-groups/invalid-dangling-groupname-4-u.js)
- [`/(?<a>\\a)/u`](https://github.com/tc39/test262/blob/0160cab5185232d0f7ada295b90fa687212a22bf/test/language/literals/regexp/named-groups/invalid-identity-escape-in-capture-u.js)


from [unicode-property-names.js](https://github.com/tc39/test262/blob/69c1efd325deedf54db92a23008399e2b00fa51e/test/built-ins/RegExp/named-groups/unicode-property-names.js)

- [`/(?<❤>a)/u`](https://github.com/tc39/test262/blob/0160cab5185232d0f7ada295b90fa687212a22bf/test/language/literals/regexp/named-groups/invalid-non-id-start-groupspecifier-u.js)
- [`/(?<𐒤>a)/u`](https://github.com/tc39/test262/blob/0160cab5185232d0f7ada295b90fa687212a22bf/test/language/literals/regexp/named-groups/invalid-non-id-start-groupspecifier-2-u.js)
- [`/(?<a\\uD801>.)/u`](https://github.com/tc39/test262/blob/0160cab5185232d0f7ada295b90fa687212a22bf/test/language/literals/regexp/named-groups/invalid-non-id-start-groupspecifier-4-u.js)
- [`/(?<a\\uDCA4>.)/u`](https://github.com/tc39/test262/blob/0160cab5185232d0f7ada295b90fa687212a22bf/test/language/literals/regexp/named-groups/invalid-non-id-start-groupspecifier-5-u.js)
- [`/(?<a\\u{110000}>.)/u`](https://github.com/tc39/test262/blob/0160cab5185232d0f7ada295b90fa687212a22bf/test/language/literals/regexp/named-groups/invalid-non-id-start-groupspecifier-9-u.js)
- [`/(?<a\uD801>.)/u`](https://github.com/tc39/test262/blob/0160cab5185232d0f7ada295b90fa687212a22bf/test/language/literals/regexp/named-groups/invalid-lone-surrogate-groupname.js)
- [`/(?<a\uDCA4>.)/u`](https://github.com/tc39/test262/blob/0160cab5185232d0f7ada295b90fa687212a22bf/test/language/literals/regexp/named-groups/invalid-lone-surrogate-groupname.js)
- `/(?<a\\uD801>.)/u` DUP IN ORIGINAL
- `/(?<a\\uDCA4>.)/u` DUP IN ORIGINAL
- [`/(?<\\>.)/u`](https://github.com/tc39/test262/blob/0160cab5185232d0f7ada295b90fa687212a22bf/test/language/literals/regexp/named-groups/invalid-non-id-start-groupspecifier-8-u.js)
- [`/(?<a\\>.)/u`](https://github.com/tc39/test262/blob/0160cab5185232d0f7ada295b90fa687212a22bf/test/language/literals/regexp/named-groups/invalid-non-id-continue-groupspecifier-4-u.js)

**Comment by bakkot** [comment]
There were four tests which were testing for regex literals which contained a lone surrogate (as opposed to an escape sequence for a lone surrogate), which I've left as `eval`-based [here](https://github.com/tc39/test262/blob/0160cab5185232d0f7ada295b90fa687212a22bf/test/language/literals/regexp/named-groups/invalid-lone-surrogate-groupname.js) because I'm not aware of any other way to represent those tests.

(I missed that there were not duplicates with the tests for literals containing an escape sequence the first time around; they are four of the nine tests my first pass missed. The other five are `/\\k<a>(?<b>x)/` and `/(?<\\u{03C0}>a)/`, which I just overlooked, `(?<\u{0041}>.)` and `(?<a\u{104A4}>.)`, which I had accidentally removed, and `/(?<\\u{0041}>.)/`, which I had typo'd as `/(?<\\u{0041}>a)/`.)

**Comment by rwaldron** [comment]
@bakkot thanks for revisiting this, and for your thorough follow up notes :D

**Pull request merged**